plugins/microservices/msi_get_agent_pid/libmsi_get_agent_pid.cpp

```diff
@@ -0,0 +1,47 @@
+#include "irods_ms_plugin.hpp"
+#include "irods_re_structs.hpp"
+#include "msParam.h"
+#include "rodsErrorTable.h"
+#include "irods_error.hpp"
+#include "irods_logger.hpp"
+
+#include <exception>
+#include <functional>
+#include <string>
+
+namespace
+{
+    using log = irods::experimental::log;
+
+    auto msi_impl(msParam_t* out_pid, ruleExecInfo_t* rei) -> int
+    {
+        try {
+            const char* pid = std::to_string(getpid()).c_str();
```

Comment:
this is effectively invoking undefined behavior because the temporary produced by `std::to_string` goes away.  you can capture the `std::string` and invoke `.c_str()` on the next call.